cognite/neat/rules/exporter/rules2excel.py

```diff
@@ -5,62 +5,108 @@
 from openpyxl.cell import Cell
 from openpyxl.styles import Alignment, Border, Font, NamedStyle, PatternFill, Side
 
-from cognite.neat.rules.models import TransformationRules
+from cognite.neat.rules.models.raw_rules import RawRules
+from cognite.neat.rules.models.rules import Rules
 
+from ._base import BaseExporter
 
-class RulesToExcel:
+
+class ExcelExporter(BaseExporter):
     """Class for exporting transformation rules object to excel file."""
 
-    def __init__(self, rules: TransformationRules):
-        self.rules = rules
-        self.workbook = Workbook()
+    def __init__(self, rules: Rules | RawRules, filepath: Path, report_path: Path | None = None):
+        self.report = None
+        if rules.__class__.__name__ == RawRules.__name__:
+            self.rules = cast(RawRules, rules).to_rules(skip_validation=True)
+            self.report = cast(RawRules, rules).validate_rules()
+        else:
+            self.rules = cast(Rules, rules)
+
+        self.filepath = filepath
+        self.report_path = report_path
         self.class_counter = 0
         self.property_counter = 0
+        self.generate_workbook()
+        super().__init__(self.rules, self.filepath, self.report_path)
+
+    def export(self, filepath: Path | None = None):
+        """Exports transformation rules to excel file."""
+        filepath = filepath or self.filepath
+        if not filepath:
+            raise ValueError("No filepath given")
 
-    @classmethod
-    def export_rules_to_file(cls, rules: TransformationRules, file_path: Path):
-        """Generates workbook from transformation rules and saves it to file."""
-        instance = cls(rules=rules)
-        instance.generate_workbook()
-        instance.save_to_file(file_path=file_path)
+        self.data.save(self.filepath)
+        self.data.close()
```

Comment:
Should have a try-finally block here to ensure that `.close()` is always called, so we don't leave the file in an invalid state.

Reply:
implemented